docker/resource_docker_plugin.go

```diff
@@ -33,9 +33,33 @@ func resourceDockerPlugin() *schema.Resource {
 				Default:  true,
 			},
 			"grant_all_permissions": {
-				Type:        schema.TypeBool,
-				Optional:    true,
-				Description: "If true, grant all permissions necessary to run the plugin",
+				Type:          schema.TypeBool,
+				Optional:      true,
+				Description:   "If true, grant all permissions necessary to run the plugin",
+				ConflictsWith: []string{"grant_permissions"},
+			},
+			"grant_permissions": {
+				Type:          schema.TypeSet,
```

Comment:
Should this be a TypeMap? keyed on the name

Reply:
TypeMap supports only primitive type.

https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/helper/schema#Schema

> For a TypeMap, it must be a *Schema with a Type that is one of the primitives: TypeString, TypeBool, TypeInt, or TypeFloat

Reply:
Ah, right. That is unfortunate.